internal/kafka/internal/workers/cluster_mgrs/dynamic_scale_down_mgr.go

```diff
@@ -0,0 +1,241 @@
+}
+
+var _ workers.Worker = &DynamicScaleDownManager{}
+
+func NewDynamicScaleDownManager(
+	reconciler workers.Reconciler,
+	dataplaneClusterConfig *config.DataplaneClusterConfig,
+	clusterProvidersConfig *config.ProviderConfig,
+	kafkaConfig *config.KafkaConfig,
+	clusterService services.ClusterService,
+) *DynamicScaleDownManager {
+
+	return &DynamicScaleDownManager{
+		BaseWorker: workers.BaseWorker{
+			Id:         uuid.New().String(),
+			WorkerType: dynamicScaleDownWorkerType,
+			Reconciler: reconciler,
+		},
+
+		dataplaneClusterConfig: dataplaneClusterConfig,
+		clusterProvidersConfig: clusterProvidersConfig,
+		kafkaConfig:            kafkaConfig,
+		clusterService:         clusterService,
+	}
+}
+
+func (m *DynamicScaleDownManager) Start() {
+	m.StartWorker(m)
+}
+
+func (m *DynamicScaleDownManager) Stop() {
+	m.StopWorker(m)
+}
+
+func (m *DynamicScaleDownManager) Reconcile() []error {
+	var errList fleeterrors.ErrorList
+	if !m.dataplaneClusterConfig.IsDataPlaneAutoScalingEnabled() {
+		glog.Infoln("dynamic scaling is disabled. Dynamic scale down reconcile event skipped")
+		return nil
+	}
+	glog.Infoln("running dynamic scale down reconcile event")
+	defer m.logReconcileEventEnd()
+
+	err := m.processDynamicScaleDownReconcileEvent()
+	if err != nil {
+		errList.AddErrors(err)
+	}
+
+	return errList.ToErrorSlice()
+}
+
+func (m *DynamicScaleDownManager) processDynamicScaleDownReconcileEvent() error {
+	var errList fleeterrors.ErrorList
+	kafkaStreamingUnitCountPerClusterList, err := m.clusterService.FindStreamingUnitCountByClusterAndInstanceType()
+	if err != nil {
+		errList.AddErrors(err)
+		return errList
+	}
+
+	var alreadyProcessedClusters map[string]bool = map[string]bool{}
+
+	for _, kafkaStreamingUnitCountPerCluster := range kafkaStreamingUnitCountPerClusterList {
+		clusterID := kafkaStreamingUnitCountPerCluster.ClusterId
+		_, alreadyProcessed := alreadyProcessedClusters[clusterID]
+		if alreadyProcessed {
+			continue
+		}
+
+		var dynamicScaleDownProcessor dynamicScaleDownProcessor = &standardDynamicScaleDownProcessor{
+			kafkaStreamingUnitCountPerClusterList: kafkaStreamingUnitCountPerClusterList,
+			supportedKafkaInstanceTypesConfig:     &m.kafkaConfig.SupportedInstanceTypes.Configuration,
+			clusterService:                        m.clusterService,
+			dryRun:                                true, // this could be coming from a config just like 'EnableDynamicScaleUpManagerScaleUpTrigger'
+			clusterID:                             clusterID,
+		}
+
+		glog.Infof("evaluating dynamic scale down for cluster '%s'", clusterID)
+		shouldScaleDown, err := dynamicScaleDownProcessor.ShouldScaleDown()
+		if err != nil {
+			errList.AddErrors(err)
+			continue
+		}
+		if shouldScaleDown {
+			glog.Infof("data plane scale down need detected for cluster '%s'", clusterID)
+			err := dynamicScaleDownProcessor.ScaleDown()
+			if err != nil {
+				errList.AddErrors(err)
+				continue
+			}
+		}
+
```

Comment:
@miguelsorianod this is heavily inspired by the scale up reconciler design.
Let me know what you think